### PR TITLE
Fix bike question

### PR DESCRIPTION
**Pull request by k-sharpe**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -63,21 +63,28 @@ function reverseHeroText(currentCharIndex, curTypewriterTextIndex) {
...
+      reGuess = window.confirm("Omg, only off by 5 miles or more. Try again, you're so close.");
+    } else if(guess >= 35 || guess <= 29) {
+      reGuess = window.confirm("WOW! Only off by 3 miles or more. Try again if you'd like.");
+    } else if (guess > 32 || guess < 32) {
+      reGuess = window.confirm("So so SO close, but not quite!");
+    } else if (guess == 32) {
+      reGuess = false;
       window.alert("Hurrah! I went exactly 32.0 miles that Thursday!");
+    }
+    else {
```
Comment (andreahu12) [inline portfolio/src/main/webapp/script.js:85]: nit: unnecessary newline

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -63,21 +63,28 @@ function reverseHeroText(currentCharIndex, curTypewriterTextIndex) {
...
+      reGuess = window.confirm("Not bad! Only off by 20 miles or more. Try again if you'd like");
+    } else if (guess >= 42 || guess <= 22) {
+      reGuess = window.confirm("Super close! Only off by 10 miles or more. Try again!");
+    } else if (guess >= 37 || guess <= 27) {
+      reGuess = window.confirm("Omg, only off by 5 miles or more. Try again, you're so close.");
+    } else if(guess >= 35 || guess <= 29) {
+      reGuess = window.confirm("WOW! Only off by 3 miles or more. Try again if you'd like.");
+    } else if (guess > 32 || guess < 32) {
+      reGuess = window.confirm("So so SO close, but not quite!");
+    } else if (guess == 32) {
```
Comment (andreahu12) [inline portfolio/src/main/webapp/script.js:81]: prefer === instead of == because it's safer

Comment (k-sharpe) [reply portfolio/src/main/webapp/script.js]: Sounds good, converted guess to a float and fixed issue in latest commit.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -63,21 +63,28 @@ function reverseHeroText(currentCharIndex, curTypewriterTextIndex) {
...
+      reGuess = window.confirm("Not bad! Only off by 20 miles or more. Try again if you'd like");
+    } else if (guess >= 42.0 || guess <= 22.0) {
+      reGuess = window.confirm("Super close! Only off by 10 miles or more. Try again!");
+    } else if (guess >= 37.0 || guess <= 27.0) {
+      reGuess = window.confirm("Omg, only off by 5 miles or more. Try again, you're so close.");
+    } else if(guess >= 35.0 || guess <= 29.0) {
+      reGuess = window.confirm("WOW! Only off by 3 miles or more. Try again if you'd like.");
+    } else if (guess > 32.0 || guess < 32.0) {
+      reGuess = window.confirm("So so SO close, but not quite!");
+    } else if (guess === 32.0) {
```
Comment (jgao23) [inline portfolio/src/main/webapp/script.js:82]: i think it would be easier to read if you defined the actual miles you rode as a const, and then took the difference between the guess and actual miles and wrote the if conditions based on the difference. then the numbers in the if statement would align to the messages you return, and if you wanted to change the actual miles you rode, you wouldn't need to change all the rest of the numbers.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -63,21 +63,28 @@ function reverseHeroText(currentCharIndex, curTypewriterTextIndex) {
...
+      reGuess = false;
       window.alert("Hurrah! I went exactly 32.0 miles that Thursday!");
+    } else {
+      reGuess = window.confirm("That's not a valid guess, your guesses should look like '4' or '12.6' not " + guess);
```
Comment (jgao23) [inline portfolio/src/main/webapp/script.js:86]: js parseFloat is interesting: https://www.geeksforgeeks.org/javascript-parsefloat-with-examples/
 you can pass in a lot of different inputs that would resolve to a float, but have you tried if you inputted guess "fifty-two"? would that break?

File: portfolio/pom.xml [hunk 5]
```diff
@@ -35,7 +35,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
-          <deploy.projectId>YOUR_PROJECT_ID_HERE</deploy.projectId>
+          <deploy.projectId>ksharpe-step-2020</deploy.projectId>
```
Comment (jgao23) [inline portfolio/pom.xml:38]: this is unrelated to fixing the bike question, so you probably should have opened a separate PR for this change. but it's fine this time to keep it here.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (k-sharpe) [reply portfolio/src/main/webapp/script.js:86]: I tried entering fifty-two which resulted in NaN. Would've been pretty cool if it converted to a number.but "52fifty-two" results in 52 which is neat.

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -63,21 +63,28 @@ function reverseHeroText(currentCharIndex, curTypewriterTextIndex) {
...
-      window.alert("Omg, only off by 5 miles or more. Try again, you're so close.");
-  } else if(guess >= 35 || guess <= 29) {
-      window.alert("WOW! Only off by 3 miles or more. Try again if you'd like.");
-  } else if (guess > 32 || guess < 32) {
-      window.alert("So so close, but not quite!");
-  } else if (guess === 32) {
+  let reGuess = true;
+  while(reGuess) {
+    guess = window.prompt("How far was it? (example: 4)");
+    guess = parseFloat(guess);
```
Comment (jgao23) [inline portfolio/src/main/webapp/script.js:69]: nit: don't re-use guess, define a new const

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (jgao23) [reply portfolio/src/main/webapp/script.js:86]: supernit: if the guess doesn't resolve to a number, maybe instead of using guess here you could use the window prompt instead of the parsed value so it won't show NaN and actually show what the user inputted

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (jgao23) [reply portfolio/src/main/webapp/script.js:82]: thoughts? or do you think the way you have it currently is more readable?

Comment (k-sharpe) [reply portfolio/src/main/webapp/script.js]: I thought I changed this, mightve not been committed properly or something. I def understand that using a single unit is a lot cleaner and easier to read than this.

**Comment by k-sharpe** [comment]
I realized I was making your suggested changes in another branch, so I moved it over and also fixed the NaN display error, sorry about that.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -63,21 +63,30 @@ function reverseHeroText(currentCharIndex, curTypewriterTextIndex) {
...
+      reGuess = window.confirm("Super close! Only off by 10 miles or more. Try again!");
+    } else if (difference >= 5) {
+      reGuess = window.confirm("Omg, only off by 5 miles or more. Try again, you're so close.");
+    } else if(difference >= 3) {
+      reGuess = window.confirm("WOW! Only off by 3 miles or more. Try again if you'd like.");
+    } else if (difference >= 1) {
+      reGuess = window.confirm("So so SO close, but not quite!");
+    } else if (difference === 0.0) {
+      reGuess = false;
       window.alert("Hurrah! I went exactly 32.0 miles that Thursday!");
```
Comment (jgao23) [inline portfolio/src/main/webapp/script.js:86]: use distance instead of hard coding 32.0

Comment (k-sharpe) [reply portfolio/src/main/webapp/script.js:86]: Nice catch, thank you
